tests/helpers/pipelines.py

```diff
@@ -20,6 +20,16 @@
 from tests.helpers.utils import get_default_logger, load_model_from_checkpoint, reset_seed
 
 
+def run_model_predict(trainer_options: dict, model, data: LightningDataModule):
+    reset_seed()
+    trainer = Trainer(**trainer_options)
+
+    predictions = trainer.predict(model, data.train_dataloader(), return_predictions=True)
+    assert len(predictions) == len(
+        data.train_dataloader()
```

Comment:
yes, this cannot be satisfied. 

If you run with 2 gpus / processes and one process does predictions on one half and the other on the other half, then in each process you get

assert len(predictions) == len(data.train_dataloader()) / 2

(this assertion will hold if the dataset is evenly divisible by batch size and number of gpus)